python/pyspark/mllib/__init__.py

```diff
@@ -23,7 +23,14 @@
 # MLlib currently needs NumPy 1.4+, so complain if lower
 
 import numpy
-if numpy.version.version < '1.4':
+ver = numpy.version.version
```

Comment:
Thanks for catching this before the official 1.10 release comes out! The following may be simpler:

~~~python
ver = [int(x) for x in numpy.version.version.split(.)[:2]]
if ver < [1, 4]:
  raise ...
~~~